### ISSUE TITLE
Ports are missing the has-links class if created in OnInitializedAsync()

### ISSUE THREAD

**Issue by b43r** [Blazor-Diagrams/Blazor.Diagrams#111]
If nodes, ports and links are created in the async method OnInitializedAsync() AFTER awaiting an async method, then the DIVs used to display the ports are missing the has-links class.
If I click on a port, or pan the whole diagram off-screen and back, the missing class is automatically added.
This has to be related with the async initialize method, because when I create some nodes with links BEFORE the await, everything works as expected.

**Comment by zHaytam** [comment]
Hello,

I will investigate & try to include it in the next version. Thanks!

**Comment by b43r** [comment]
Hi,

I can reproduce the issue with your SimpleDemo example if I move the await Task.Delay(5000); at the top of the method.

**Comment by zHaytam** [comment]
Sorry I thought I have answered this...
Do you mean like this?

```cs
protected override async Task OnInitializedAsync()
{
    await Task.Delay(5000);

    var node1 = NewNode(50, 50);
    var node2 = NewNode(300, 300);
    var node3 = NewNode(300, 50);
    diagram.Nodes.Add(new[] { node1, node2, node3 });

    diagram.Links.Add(new LinkModel(node1.GetPort(PortAlignment.Right), node2.GetPort(PortAlignment.Left))
    {
        SourceMarker = LinkMarker.Arrow,
        TargetMarker = LinkMarker.Arrow
    });
    diagram.Links.Add(new LinkModel(node2.GetPort(PortAlignment.Right), node3.GetPort(PortAlignment.Right))
    {
        Router = Routers.Orthogonal,
        PathGenerator = PathGenerators.Straight,
        SourceMarker = LinkMarker.Arrow,
        TargetMarker = LinkMarker.Arrow
    });
}
```

Because this also doesn't reproduce it... Can you please show me your code?

**Comment by b43r** [comment]
Yes, that is exactly the code I used. I downloaded the latest Blazor.Diagrams-2.1.3 release and replaced the OnInitialized method in SharedDemo/Demos/Simple.razor.cs with the above code.
Ports are not drawn filled with black color, as you can see in the attached screenshot. Same behaviour with Chrome, Firefox and Edge.

![grafik](https://user-images.githubusercontent.com/37931563/130359317-d6c2e495-27fa-430f-ac6f-76de4329d24a.png)

**Comment by b43r** [comment]
Ok, I did some more research and found out that PortRenderer.BuildRenderTree() is called as soon as the nodes are added to the diagram, but BEFORE the links were actually added.
I then changed the code so that links are added before the nodes, and suddenly ports were rendered correctly:

```
        protected override async Task OnInitializedAsync()
        {
            await Task.Delay(5000);

            var node1 = NewNode(50, 50);
            var node2 = NewNode(300, 300);
            var node3 = NewNode(300, 50);

            diagram.Links.Add(new LinkModel(node1.GetPort(PortAlignment.Right), node2.GetPort(PortAlignment.Left))
            {
                SourceMarker = LinkMarker.Arrow,
                TargetMarker = LinkMarker.Arrow
            });
            diagram.Links.Add(new LinkModel(node2.GetPort(PortAlignment.Right), node3.GetPort(PortAlignment.Right))
            {
                Router = Routers.Orthogonal,
                PathGenerator = PathGenerators.Straight,
                SourceMarker = LinkMarker.Arrow,
                TargetMarker = LinkMarker.Arrow
            });

            diagram.Nodes.Add(new[] { node1, node2, node3 });
        }
```

**Comment by zHaytam** [comment]
> Yes, that is exactly the code I used. I downloaded the latest Blazor.Diagrams-2.1.3 release and replaced the OnInitialized method in SharedDemo/Demos/Simple.razor.cs with the above code.
> Ports are not drawn filled with black color, as you can see in the attached screenshot. Same behaviour with Chrome, Firefox and Edge.
> 
> In addition there was a null reference exception in NavigatorWidget.razor because NodePositionAdjustment was NULL.
> 
> ![grafik](https://user-images.githubusercontent.com/37931563/130359317-d6c2e495-27fa-430f-ac6f-76de4329d24a.png)

Was this in Server side or Wasm?

**Comment by b43r** [comment]
Interesting, the issue only happens in the Wasm version!

**Comment by zHaytam** [comment]
After more research, this seems to be an issue with parallel refreshes. I'm still looking to see if I can fix it from my side.
